static/js/publisher/release/components/revisionsListRowProgressive.js

```diff
@@ -0,0 +1,186 @@
+import React from "react";
+import PropTypes from "prop-types";
+import { connect } from "react-redux";
+
+import {
+  releaseRevision,
+  updateProgressiveReleasePercentage,
+  pauseProgressiveRelease,
+  resumeProgressiveRelease
+} from "../actions/pendingReleases";
+
+import { getProgressiveState } from "../selectors";
+
+import { InteractiveProgressiveBar } from "./progressiveBar";
+
+const RevisionsListRowProgressive = ({
+  channel,
+  architecture,
+  revision,
+  setDraggable,
+  getProgressiveState,
+  releaseRevision,
+  updateProgressiveReleasePercentage,
+  pauseProgressiveRelease,
+  resumeProgressiveRelease
+}) => {
+  const [
+    progressiveState,
+    previousRevision,
+    pendingProgressiveState
+  ] = getProgressiveState(channel, architecture, revision.revision);
+
+  let showProgressivePause = false;
+  let showProgressiveResume = false;
+
+  if (progressiveState) {
+    if (progressiveState.paused) {
+      showProgressiveResume = true;
+    } else {
+      showProgressivePause = true;
+    }
+  }
+
+  if (pendingProgressiveState) {
+    if (pendingProgressiveState.paused) {
+      showProgressivePause = false;
+      showProgressiveResume = true;
+    } else {
+      showProgressivePause = true;
+      showProgressiveResume = false;
+    }
+  }
+
+  const handlePauseProgressiveRelease = () => {
+    releaseRevision(revision, channel, progressiveState);
+    pauseProgressiveRelease(progressiveState.key);
+  };
+
+  const handleResumeProgressiveRelease = () => {
+    if (!pendingProgressiveState) {
+      releaseRevision(revision, channel, progressiveState);
+    }
+    resumeProgressiveRelease(progressiveState.key);
+  };
+
+  const handleCancelProgressiveRelease = () => {
+    releaseRevision(previousRevision, channel, null);
+  };
+
+  const handleProgressiveChange = percentage => {
+    if (!pendingProgressiveState) {
+      releaseRevision(revision, channel, progressiveState);
+    }
+    updateProgressiveReleasePercentage(progressiveState.key, percentage);
+  };
+
+  return (
+    <td>
```

Comment:
Dropping the `<td>` from here (leaving it on row component level) would make it reusable anywhere outside of the table. Unless it depends on being in the table cell (which I don't think is the case).